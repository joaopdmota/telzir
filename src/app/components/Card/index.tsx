import React, { FunctionComponent } from 'react';
import { Card, Grid } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import styled from 'styled-components';

import { H1 } from '../Layout';

const useStyles = makeStyles({
    root: {
        minWidth: 275,
        margin: 20,
        borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) 0 0 !important',
        boxShadow: '0 0.25rem 0.75rem rgba(0, 0, 0, .05) !important',
        border: '1px solid rgba(0,0,0,.125)',
    },
});

interface CardProps {
    content: any;
    title: string;
}
const StyledCard: FunctionComponent<CardProps> = ({ content, title }) => {
    const classes = useStyles();

    return (
        <Card className={classes.root}>
            <H1>{title}</H1>
            <hr />
            <Grid
                container
                direction="row"
                justify="center"
                alignItems="center"
            >
                <CardContent>{content}</CardContent>
            </Grid>
        </Card>
    );
};
const CardContent = styled.span`
    font-size: 40px;
    padding: 15px;
`;

export default StyledCard;
