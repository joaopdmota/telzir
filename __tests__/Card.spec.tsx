import React from 'react';
import { shallow } from 'enzyme';
import Card from '../src/app/components/Card';

it('should renders card', () => {
    const wrapper = shallow(<Card content="lorem ipsum" title="lorem ipsum" />);

    expect(wrapper).toMatchSnapshot();
});
