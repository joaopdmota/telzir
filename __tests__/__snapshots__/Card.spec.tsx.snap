// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders card 1`] = `ShallowWrapper {}`;
